### PR TITLE
Make conversation and inbox pages update using AJAX

### PR DESCRIPTION
**Pull request by quis**
Fairly self-explanatory. Uses the same pattern of breaking things up into functions as the jobs page.

### CODE REVIEW

File: app/main/views/dashboard.py [hunk 1]
```diff
@@ -143,6 +143,23 @@ def monthly(service_id):
...
+    return render_template(
+        'views/dashboard/inbox.html',
+        partials=get_inbox_partials(service_id),
+        updates_url=url_for('.inbox_updates', service_id=service_id),
+    )
+
+
+@main.route("/services/<service_id>/inbox.json")
+@login_required
+@user_has_permissions('view_activity', admin_override=True)
```
Comment (leohemsted) [inline app/main/views/dashboard.py]: should also add tests for using the right permissions to access `inbox` and `inbox.json`

File: tests/app/main/views/test_dashboard.py [hunk 2]
```diff
@@ -198,6 +199,30 @@ def test_inbox_not_accessible_to_service_without_permissions(
...
 
 
+def test_view_inbox_updates(
+    logged_in_client,
+    service_one,
+    mocker,
+    mock_get_inbound_sms_with_no_messages,
+):
+
+    service_one['permissions'] = ['inbound_sms']
```
Comment (leohemsted) [inline tests/app/main/views/test_dashboard.py]: `service_one` is just a json blob, so won't actually update the values in `current_service` i think?

File: app/main/views/dashboard.py (hunk 1 above)
Comment (quis) [inline app/main/views/dashboard.py]: Permissions fix is in this PR: https://github.com/alphagov/notifications-admin/pull/1361

File: tests/app/main/views/test_dashboard.py (hunk 2 above)
Comment (quis) [inline tests/app/main/views/test_dashboard.py]: giphy-1

Comment (leohemsted) [inline tests/app/main/views/test_dashboard.py]: essentially, if you put a pdb inside the actual function, does `current_service['permissions'] == ['inbound_sms']`

Comment (quis) [inline tests/app/main/views/test_dashboard.py]: Yes. Somehow. 😬

File: app/main/views/dashboard.py (hunk 1 above)
Comment (quis) [inline app/main/views/dashboard.py]: Permissions test for the json endpoint added in 813c348c

File: app/main/views/dashboard.py [hunk 3]
```diff
@@ -143,6 +143,25 @@ def monthly(service_id):
 @user_has_permissions('view_activity', admin_override=True)
 def inbox(service_id):
 
+    print(current_service)
```
Comment (leohemsted) [inline app/main/views/dashboard.py]: 🖨

Comment (quis) [inline app/main/views/dashboard.py]: It’s gone.

**Pull request merged**